Plugins/Json/src/GridJsonConverter.cpp

```diff
@@ -43,130 +44,256 @@ nlohmann::json Acts::AxisJsonConverter::toJsonDetray(const IAxis& ia) {
+                    twoDimSubspaces);
+    if (!jGlobalToGridLocal.empty()) {
+      return jGlobalToGridLocal;
+    }
   }
-
-
-
-  return jga;
+  return jGlobalToGridLocal;
 }
 
 std::unique_ptr<Acts::GridAccess::IGlobalToGridLocal>
 Acts::GridAccessJsonConverter::globalToGridLocalFromJson(
-    const nlohmann::json& jga) {
-  std::unique_ptr<Acts::GridAccess::IGlobalToGridLocal> ga;
-  return ga;
+    const nlohmann::json& jGlobalToGridLocal) {
+  std::unique_ptr<Acts::GridAccess::IGlobalToGridLocal> globalToGridLocal =
+      nullptr;
+
+  std::vector<BinningValue> accessors =
+      jGlobalToGridLocal.at("accessors").get<std::vector<BinningValue>>();
+
+  // Switch and fill for 1D
+  if (accessors.size() == 1u) {
+    switch (accessors[0]) {
+      case binX:
+        globalToGridLocal = decodeSubspace<binX>(jGlobalToGridLocal);
+        break;
+      case binY:
+        globalToGridLocal = decodeSubspace<binY>(jGlobalToGridLocal);
+        break;
+      case binZ:
+        globalToGridLocal = decodeSubspace<binZ>(jGlobalToGridLocal);
+        break;
+      case binR:
+        globalToGridLocal = decodeSubspace<binR>(jGlobalToGridLocal);
+        break;
+      case binPhi:
+        globalToGridLocal = decodeSubspace<binPhi>(jGlobalToGridLocal);
+        break;
+      case binEta:
+        globalToGridLocal = decodeSubspace<binEta>(jGlobalToGridLocal);
+        break;
+      default:
+        break;
+    }
+  }
+
+  // Switch and fill for 2D
+  if (accessors.size() == 2u) {
+    if (accessors == std::vector<BinningValue>{binX, binY}) {
+      globalToGridLocal = decodeSubspace<binX, binY>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binY, binX}) {
+      globalToGridLocal = decodeSubspace<binY, binX>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binX, binZ}) {
+      globalToGridLocal = decodeSubspace<binX, binZ>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binX}) {
+      globalToGridLocal = decodeSubspace<binZ, binX>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binY, binZ}) {
+      globalToGridLocal = decodeSubspace<binY, binZ>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binY}) {
+      globalToGridLocal = decodeSubspace<binZ, binY>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binR, binPhi}) {
+      globalToGridLocal = decodeSubspace<binR, binPhi>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binPhi, binR}) {
+      globalToGridLocal = decodeSubspace<binPhi, binR>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binPhi}) {
+      globalToGridLocal = decodeSubspace<binZ, binPhi>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binPhi, binZ}) {
+      globalToGridLocal = decodeSubspace<binPhi, binZ>(jGlobalToGridLocal);
+    }
```

Comment:
```suggestion
    }
    // else globalToGridLocal = nullptr;
```

Reply:
It should be set to nullptr already - is this just for readability?

Reply:
Yes, just for readability, since it is set quite a few lines above.

I thought about adding a `throw` instead of the `break` but just returning `nullptr` could be fine as well.